### PR TITLE
Provider modal and bugs fixes

### PR DESCRIPTION
**Pull request by LiorAgnin**
## Proposed changes

Features:
- Added provider modal

Bugs fixed:
- transfer to funder form
- Reward user form
- Mixpanel events
- Update balances on transfers

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: dapp/src/components/dashboard/components/Sidebar/index.jsx [hunk 1]
```diff
@@ -196,6 +196,14 @@ const Sidebar = ({ communityName, match, isAdmin, isGradientLogo, plugins, token
     return () => {}
   }, [isAdmin, tokenType])
 
+  useEffect(() => {
+    const paramsArr = location.pathname.split('/')
+    const lastItem = paramsArr[paramsArr.length - 1]
+    if ((communityAddress !== lastItem) && lastItem !== 'justCreated') {
```
Comment (leonprou) [inline dapp/src/components/dashboard/components/Sidebar/index.jsx]: that something for analytics? You set it after the issuance?

File: dapp/src/components/dashboard/pages/JoinBonus/index.jsx [hunk 2]
```diff
@@ -36,29 +33,26 @@ const JoinBonus = ({
 }) => {
   useSwitchNetwork(networkType, 'join bonus')
 
-  const { plugins } = community
+  const funderAccount = useSelector(getFunderAccount)
```
Comment (leonprou) [inline dapp/src/components/dashboard/pages/JoinBonus/index.jsx]: that's like `mapStateToProps`  right?

File: server/src/routes/api/payments.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const router = require('express').Router()
```
Comment (leonprou) [inline server/src/routes/api/payments.js]: you can delete this from  your branch :)

File: dapp/src/components/dashboard/pages/JoinBonus/index.jsx (hunk 2 above)
Comment (LiorAgnin) [reply dapp/src/components/dashboard/pages/JoinBonus/index.jsx]: Yes.

File: server/src/routes/api/payments.js (hunk 3 above)
Comment (LiorAgnin) [reply server/src/routes/api/payments.js]: NP

File: dapp/src/components/dashboard/components/Balance/index.jsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  token,
+  balances,
+  openModal,
+  balanceOfToken,
+  fetchTokenTotalSupply
+}) => {
+  const { symbol } = token
+  const { bridge } = bridgeSide
+  const balance = balances[tokenAddress]
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (LiorAgnin) [reply dapp/src/components/dashboard/components/Balance/index.jsx]: Good idea,  I'll change that.

File: dapp/src/components/dashboard/components/Bridge/index.jsx [hunk 5]
```diff
@@ -276,41 +191,61 @@ Bridge.propTypes = {
...
+  return (
+    <Bridge
+      {...props}
+      waitingForConfirmation={isWaitingForConfirmation()}
+      transferStatus={getTransferStatus()}
+      waitingForRelayEvent={getTransferStatus() === 'WAITING FOR BRIDGE'}
+    />
+  )
+}, (prevProps, nextProps) => {
+  if (prevProps.balances[prevProps.community.homeTokenAddress] !== nextProps.balances[nextProps.community.homeTokenAddress]) {
```
[the remark this answers is not in the document]
Comment (LiorAgnin) [reply dapp/src/components/dashboard/components/Bridge/index.jsx]: No once.
Functional components renders every time that his father re-render, which happens a lot with his father

File: dapp/src/components/dashboard/components/Sidebar/index.jsx (hunk 1 above)
Comment (LiorAgnin) [reply dapp/src/components/dashboard/components/Sidebar/index.jsx]: No, it's for setting the path in the initialization

**Pull request merged**